catalog/spatial/registry/registry-source-configuration-handler/src/main/java/org/codice/ddf/registry/source/configuration/SourceConfigurationHandler.java

```diff
@@ -0,0 +1,680 @@
+            }
+
+        } catch (FederationAdminException e) {
+            LOGGER.error(
+                    "Error getting registry metacards. Registry source configurations won't be updated.");
+        }
+    }
+
+    public void setActivateConfigurations(Boolean activateConfigurations) {
+        if (!activateConfigurations.equals(this.activateConfigurations)) {
+            this.activateConfigurations = activateConfigurations;
+
+            if (!activateConfigurations || preserveActiveConfigurations == null
+                    || preserveActiveConfigurations) {
+                return;
+            }
+
+            updateRegistrySourceConfigurations();
+        }
+    }
+
+    public void setPreserveActiveConfigurations(Boolean preserveActiveConfigurations) {
+        if (!preserveActiveConfigurations.equals(this.preserveActiveConfigurations)) {
+            this.preserveActiveConfigurations = preserveActiveConfigurations;
+
+            if (preserveActiveConfigurations || activateConfigurations == null
+                    || !activateConfigurations) {
+                return;
+            }
+
+            updateRegistrySourceConfigurations();
+        }
+    }
+
+    public void setCleanUpOnDelete(Boolean cleanUpOnDelete) {
+        this.cleanUpOnDelete = cleanUpOnDelete;
+    }
+
+    public void setBindingTypeFactoryPid(List<String> bindingTypeFactoryPid) {
+        bindingTypeToFactoryPidMap = new HashMap<>();
+
+        for (String mapping : bindingTypeFactoryPid) {
+            String bindingType = StringUtils.substringBefore(mapping, "=");
+            String factoryPid = StringUtils.substringAfter(mapping, "=");
+
+            if (StringUtils.isNotBlank(bindingType) && StringUtils.isNotBlank(factoryPid)) {
+                bindingTypeToFactoryPidMap.put(bindingType, factoryPid);
+            }
+        }
+    }
+
+    public void setSourceActivationPriorityOrder(List<String> sourceActivationPriorityOrder) {
+        if (sourceActivationPriorityOrder != null
+                && !sourceActivationPriorityOrder.equals(this.sourceActivationPriorityOrder)) {
+            this.sourceActivationPriorityOrder = sourceActivationPriorityOrder;
+
+            if (activateConfigurations == null || preserveActiveConfigurations == null
+                    || !activateConfigurations && preserveActiveConfigurations) {
+                return;
+            }
+
+            updateRegistrySourceConfigurations();
+        }
+    }
+
+    public void setUrlBindingName(String urlBindingName) {
+        this.urlBindingName = urlBindingName;
+    }
+
+    public void setConfigurationAdmin(ConfigurationAdmin configurationAdmin) {
+        this.configurationAdmin = configurationAdmin;
+    }
+
+    public void setMetaTypeService(MetaTypeService metaTypeService) {
+        this.metaTypeService = metaTypeService;
+    }
+
+    public void setParser(Parser parser) {
+        List<String> contextPath = Arrays.asList(RegistryObjectType.class.getPackage()
```

Comment:
We could maybe pull the context path part into a utility method but we couldn't abstract the setParser method without having a bunch of unrelated classes extend a common abstract class. Since this change would affect multiple packages I think we will create a separate ticket for it.